pymc/distributions/shape_utils.py

```diff
@@ -471,19 +473,18 @@ def convert_size(size: Size) -> Optional[StrongSize]:
     """Process a user-provided size variable into None or a valid size object."""
     if size is None:
         return None
-
-    if isinstance(size, int) or (isinstance(size, TensorVariable) and size.ndim == 0):
-        size = (size,)
+    if isinstance(size, int):
+        size = size
+    elif isinstance(size, TensorVariable) and size.ndim <= 1:
+        size = tuple(size)
```

Comment:
`tuple(size)` does't work when `size.ndim == 0`.
The 0 and 1 ndim must be separate cases

Reply:
does `to_tuple` work?

Reply:
something like this would be sufficient ?
```
if isinstance(shape, int) or (isinstance(size,TensorVariable) and shape.ndim==0):
    shape = shape
if isinstance(shape, TensorVariable) and shape.ndim == 1:
    shape = tuple(shape)
```

Reply:
Just keep the code that was there before and add another elif case:
```
if isinstance(size, int) or (isinstance(size, TensorVariable) and size.ndim == 0):
    size = (size,)
elif isinstance(size, TensorVariable) and size.ndim == 1:
    size = tuple(size)
```
And the same further up in `convert_shape`

Reply:
on using 
```
pre-commit run --all
```
all the tests are passed, however it shows build failed when I commit the changes how can I resolve this?

Reply:
Precommit is not running tests, but other things like linters and code format checks. The output should tell you what the fix is, or automatically add the changes to git staging. Feel free to post a screenshot of the output and we can more specifically let you know what to do!

Reply:
<img width="1016" alt="image" src="https://user-images.githubusercontent.com/17161999/158074111-382bf0f6-82e8-4cf7-b2aa-2c4a10375b4c.png">
this shows pre commit works fine, yet the commit fails to build.
How can I fix this?

Reply:
Im not sure what you mean by build. This indicates the commit should have taken place. What happens when you try and push it to github?

Reply:
I think your branch could be outdated. The [pre-commit in the CI failed in the type checking step](https://github.com/pymc-devs/pymc/runs/5528156814?check_suite_focus=true#step:4:64) but that step did not run in your screenshot.

Do the following to update your branch:
1. Make sure that you don't have uncommitted changes (commit, stash or revert them if needed)
2. `git checkout main`
3. `git pull`
4. `git checkout my-feature` to go back to the branch of this PR
5. `git rebase main` to rebase the PR branch onto main. Since GitHub doesn't show any merge conflicts this step should work without problems.

Then run `pre-commit` again. This time it should reproduce the failure from the CI.